components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -258,4 +295,245 @@ private boolean canAuthenticate(OAuthClientAuthenticator oAuthClientAuthenticato
+            String oauthJWTAssertion = request.getParameter(OAuthConstants.OAUTH_JWT_ASSERTION);
+            String oauthJWTAssertionType = request.getParameter(OAuthConstants.OAUTH_JWT_ASSERTION_TYPE);
+            String authorizationHeaderValue = request.getHeader(OAuthConstants.AUTHORIZATION_HEADER);
+            String x509Certificate = request.getHeader(certificate);
+            if (StringUtils.isEmpty(x509Certificate)) {
+                x509Certificate = IdentityUtil.convertCertificateToPEM((Certificate) request
+                        .getAttribute(certificate));
+            }
+            return (StringUtils.isNotEmpty(oauthClientID) && StringUtils.isEmpty(oauthClientSecretValue) &&
+                    StringUtils.isEmpty(oauthJWTAssertion) && StringUtils.isEmpty(oauthJWTAssertionType) &&
+                    StringUtils.isEmpty(authorizationHeaderValue) && x509Certificate != null &&
+                    OAuth2Util.parseCertificate(x509Certificate) != null);
+        } catch (CertificateException e) {
+            throw new OAuthClientAuthnException("Error occurred while parsing the certificate",
+                    OAuth2ErrorCodes.INVALID_REQUEST, e);
+        }
+    }
+
+    /**
+     * Validate whether the request follows Private Key JWT client authentication.
+     *
+     * @param request       Http servlet request.
+     * @return Whether the request follows Private Key JWT client authentication.
+     */
+    public boolean isPrivateKeyJWTAuthentication(HttpServletRequest request) {
+
+        String oauthJWTAssertionType = request.getParameter(OAuthConstants.OAUTH_JWT_ASSERTION_TYPE);
+        String oauthJWTAssertion = request.getParameter(OAuthConstants.OAUTH_JWT_ASSERTION);
+        return OAuthConstants.OAUTH_JWT_BEARER_GRANT_TYPE.equals(oauthJWTAssertionType) &&
+                StringUtils.isNotEmpty(oauthJWTAssertion);
+    }
+
+    /**
+     * Validate whether a TLS certificate is passed through the request.
+     *
+     * @param request     Http servlet request.
+     * @return Whether a TLS certificate is passed through the request.
+     * @throws OAuthClientAuthnException
+     */
+    private boolean isMTLSEnforced(HttpServletRequest request) throws OAuthClientAuthnException {
+
+        String mtlsAuthHeader = Optional.ofNullable(IdentityUtil.getProperty(OAuthConstants.MTLS_AUTH_HEADER))
+                .orElse("CONFIG_NOT_FOUND");
+        String x509Certificate = request.getHeader(mtlsAuthHeader);
+        try {
+            if (!(StringUtils.isNotEmpty(x509Certificate) &&
+                    OAuth2Util.parseCertificate(x509Certificate) != null)) {
+                log.error("Transport certificate not passed through the request or the certificate is not valid");
+                return false;
+            }
+        } catch (CertificateException e) {
+            log.error("Invalid transport certificate.", e);
+            return false;
+        }
+        return true;
```

Comment:
Here we are returning true even if it returned the default value "CONFIG_NOT_FOUND". In that case, x509Certificate can be empty.

Reply:
When the x509Certificate is empty it will be captured in the if condition prior to that return statement and handled accordingly. If the flow reaches the final return statement it means that the certificate is present and valid.